src/api/observable.ts

```diff
@@ -116,6 +116,35 @@ export class IObservableFactories {
 		return res as any;
 	}
 
+	dynamic<T>(props: T, name?: string): T & IObservableObject {
```

Comment:
Maybe check of the props object is already wrapped by a dynamic observable Proxy. Accidental double wrapping could cause some nasty bugs.